db-engine/src/main/java/com/splicemachine/db/iapi/types/SQLChar.java

```diff
@@ -1179,8 +1124,169 @@ private void resetForMaterialization() {
+     +---+---+---+---+---+---+---+---+
+     third byte is:
+     +---+---+---+---+---+---+---+---+
+     | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |
+     +---+---+---+---+---+---+---+---+
+
+
+     The UTF format:
+     Writes a string to the underlying output stream using UTF-8
+     encoding in a machine-independent manner.
+     <p>
+     First, two bytes are written to the output stream as if by the
+     <code>writeShort</code> method giving the number of bytes to
+     follow. This value is the number of bytes actually written out,
+     not the length of the string. Following the length, each character
+     of the string is output, in sequence, using the UTF-8 encoding
+     for the character.
+     @exception  IOException  if an I/O error occurs.
+     @since      JDK1.0
+
+
+     @exception IOException thrown by writeUTF
+
+     @see java.io.DataInputStream
 
-    public void readExternal(ObjectInput in) throws IOException {
+     */
+    @Override
+    protected void writeExternalOld(ObjectOutput out) throws IOException {
+        out.writeBoolean(isNull);
+
+        if (isNull()) {
+            return;
+        }
+
+        //
+        // This handles the case that a CHAR or VARCHAR value was populated from
+        // a user Clob.
+        //
+        if ( _clobValue != null )
+        {
+            writeClobUTF( out );
+            return;
+        }
+
+        String lvalue = null;
+        char[] data = null;
+
+        int strlen = rawLength;
+        boolean isRaw;
+
+        if (strlen < 0) {
+            lvalue = value;
+            strlen = lvalue.length();
+            isRaw = false;
+        } else {
+            data = rawData;
+            isRaw = true;
+        }
+
+        // byte length will always be at least string length
+        int utflen = strlen;
+
+        for (int i = 0 ; (i < strlen) && (utflen <= 65535); i++)
+        {
+            int c = isRaw ? data[i] : lvalue.charAt(i);
+            if ((c >= 0x0001) && (c <= 0x007F))
+            {
+                // 1 byte for character
+            }
+            else if (c > 0x07FF)
+            {
+                utflen += 2; // 3 bytes for character
+            }
+            else
+            {
+                utflen += 1; // 2 bytes for character
+            }
+        }
+
+        StreamHeaderGenerator header = getStreamHeaderGenerator();
+        if (SanityManager.DEBUG) {
+            SanityManager.ASSERT(!header.expectsCharCount());
+        }
+        // Generate the header, write it to the destination stream, write the
+        // user data and finally write an EOF-marker is required.
+        header.generateInto(out, utflen);
+        writeUTF(out, strlen, isRaw, null );
+        header.writeEOF(out, utflen);
+    }
+
+    @Override
+    protected void readExternalNew(ObjectInput in) throws IOException {
+        byte[] bs = ArrayUtil.readByteArray(in);
+        ExtensionRegistry extensionRegistry = ExtensionRegistry.newInstance();
```

Comment:
is the ExtensionRegistry re-created for every readXNew? is that efficient? shouldn't we maybe have a ExtensionRegistry extensionRegistry as a parameter to readExternalNew, and have these things intialized once?

Reply:
Good point. Fixed

Reply:
i don't see this being resolved, am i missing something?